dao/users.js

```diff
@@ -0,0 +1,108 @@
+const debug = require('debug')('calendar-server:reminders');
+const bcrypt = require('bcrypt');
+
+const database = require('./database');
+const { NotFoundError } = require('../utils/errors');
+const {
+  checkPropertyType, checkIsArray
+} = require('../utils/object_validator.js');
+
+function notFoundError(id) {
+  return NotFoundError.createWithSubject('user', { name: 'id', value: id });
+}
+
+const BCRYPT_SALT_ROUNDS = 10;
+const SHA_HASH_SALT = 'todochangeme';
```

Comment:
that said, looking at the lib API, I'm not convinced we're talking about the same type of salt (as in: `result = salt + bcrypt(salt + password)`).
I'll dig deeper.